### ISSUE TITLE
v0.21.0 testing

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#20555]
Link to announcement:
* rc5: https://lists.linuxfoundation.org/pipermail/bitcoin-core-dev/2021-January/000096.html
* rc3: https://lists.linuxfoundation.org/pipermail/bitcoin-core-dev/2020-December/000095.html
* rc2: https://lists.linuxfoundation.org/pipermail/bitcoin-core-dev/2020-November/000094.html

Let us know which version you tested on which operating system.

If you find an issue, please search Github for known issues first and then open a new Github issue. This meta issue should not be used to report bugs, as a single thread makes it impossible to track more than one topic.

See [Release Notes](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.21.0-Release-Notes-Draft) for a list of changes, and testing reports for earlier releases ([v0.19.0](https://github.com/bitcoin/bitcoin/issues/17079)), for an idea what to test.

Release schedule: #18947

**Comment by hebasto** [comment]
v0.21.0rc2 on ODROID-HC2 (arm7) + Armbian (based on Focal) without GUI -- works fine.

v0.21.0rc2 on Linux Mint 20, Tor-only mode -- no incoming connections to my v3 hidden service since 2020-11-27

**Comment by jonatack** [comment]
I'll happy to addnode to anyone who is testing 0.21 with tor v3; currently I know of a half dozen v3 peers. Ping me on IRC.

**Comment by decryp2kanon** [comment]
v0.21.0rc1 on Ryzen + Debian 10 (buster), normal GUI -- works fine include IBD

**Comment by wodry** [comment]
0.21.0rc2-x86_64-linux-gnu Bitcoin-QT on Debian 10 (Buster) (Intel KVM) starts and runs ok. No inbound connections with onlynet=onion to v3 Onion service.

**Comment by jonatack** [comment]
0.21rc2 on Debian 5.9.11-1 (2020-11-27) x86_64 GNU/Linux running tor v3 `70016/Satoshi:0.21.0(jon)/`, initial testing good so far, will edit/update this comment after more time and testing.

**Comment by jarolrod** [comment]
Tested (0.21rc2) torv3, signet, and gui IBD on Arch Linux 5.8.16. Tested gui and IBD on FreeBSD 12.2, and  macOS 10.15.7. Everything I tested works fine.
@hebasto I was able to connect to your (among other) Torv3 nodes in my testing.

**Comment by GhostOfSatoshi** [comment]
v0.21.0rc2 on Win2016 (up to date)
Works fine so far (without Tor)

**Comment by mjdietzx** [comment]
0.21.0rc2-osx.dmg downloaded binary works on macOS Big Sur (Version 11.0.1) ✅

**Comment by jarolrod** [comment]
> 0.21.0rc2-osx.dmg downloaded binary works on macOS Big Sur (Version 11.0.1) white_check_mark

There's a new release candidate ([rc3](https://github.com/bitcoin/bitcoin/releases/tag/v0.21.0rc3)) and [binaries](https://bitcoincore.org/bin/bitcoin-core-0.21.0/test.rc3/)

**Comment by jonatack** [comment]
0.21.0rc3 downloaded binary running on Debian 5.9.11-1 (2020-11-27) x86_64 GNU/Linux running tor v3

**Comment by mjdietzx** [comment]
> > 0.21.0rc2-osx.dmg downloaded binary works on macOS Big Sur (Version 11.0.1) white_check_mark
> 
> There's a new release candidate ([rc3](https://github.com/bitcoin/bitcoin/releases/tag/v0.21.0rc3)) and [binaries](https://bitcoincore.org/bin/bitcoin-core-0.21.0/test.rc3/)

🚨 0.21.0rc3-osx.dmg downloaded binary, **does not work** on macOS Big Sur (Version 11.0.1). When it opens: `"Bitcoin Core" is damaged and can't be opened. You should move it to the trash`

**Comment by achow101** [comment]
> rotating_light 0.21.0rc3-osx.dmg downloaded binary, **does not work** on macOS Big Sur (Version 11.0.1). When it opens: `"Bitcoin Core" is damaged and can't be opened. You should move it to the trash`

This is a known issue and we are actively ~~beating our heads against Apple's walled garden wall~~ debugging this issue. See the super long discussion that we are having on [IRC](http://gnusha.org/bitcoin-core-dev/2020-12-11.log), starting around 11:09

**Comment by jsarenik** [comment]
rc3: `0.21.0rc3-x86_64-linux-gnu.tar.gz` mainnet, testnet, signet (all with latest `master` C-Lightning on top)

**Comment by sangaman** [comment]
I'm using rc3 on an arm7 ODROID HC1 on Ubunut 18.04 with lnd & electrs and `peercfilters` enabled, so far so good.

**Comment by philipglazman** [comment]
0.21.0rc3 for Alpine Linux works fine

**Comment by 0xB10C** [comment]
0.21.0rc3 compiled from source, Ubuntu 20.04 + [i3](https://i3wm.org/): tested mainnet, testnet, and signet. created a descriptor wallet in qt, tested torv3 with addnode, and checked if anchors.dat is created on shutdown and removed on startup.

I've noticed the "**Advanced optio**ns" label being cut-off in the wallet creation dialog. 
![image](https://user-images.githubusercontent.com/19157360/102551621-1713bc80-40c0-11eb-8f4d-8e3e3576cd7e.png)

**Comment by jonatack** [comment]
@0xB10C pinged me and I verified not seeing the text issue on rc3 binary or master from source (Debian)

**Comment by hebasto** [comment]
@0xB10C 
> 0.21.0rc3 compiled from source, Ubuntu 20.04 + [i3](https://i3wm.org/): tested mainnet, testnet, and signet. created a descriptor wallet in qt, tested torv3 with addnode, and checked if anchors.dat is created on shutdown and removed on startup.
> 
> I've noticed the "**Advanced optio**ns" label being cut-off in the wallet creation dialog.

Mind testing https://github.com/bitcoin-core/gui/pull/155 ?

**Comment by dhruv** [comment]
Tested using [these instructions](https://github.com/adamjonas/bitcoin-core-rc-testing/blob/121720-testing-instructions-refinement/docs/v0.21/testing-0.21rc2.md)

Hardware: MacBook Pro (13-inch, 2018)
OS: macOS Catalina 10.15.7 (19H2)
RC: bitcoin-0.21.0rc3

Tests: 
- Created a descriptor wallet(loaded the sqlite file and ran some sql queries)
- Ran tor via docker and connected over Torv3
- Connected to common signet (saw one connection; used faucet; saw unconfirmed balance)
- Ran through the anchors test(checked anchors.dat created with clean shutdown, disappears on startup; also that it is not created with unclean shutdown)

**Comment by glozow** [comment]
0.21.0rc3 compiled from source, macbook pro 2019, macOS Catalina 10.15.7

also went through [this doc](https://github.com/adamjonas/bitcoin-core-rc-testing/blob/121720-testing-instructions-refinement/docs/v0.21/testing-0.21rc2.md) - descriptor wallets, torv3, signet, anchors worked nicely

**Comment by jamaljsr** [comment]
0.21.0rc3 compiled from source on MacBook Pro (16-inch, 2019) running MacOS Big Sur v11.0.1

Followed @adamjonas [test script](https://github.com/adamjonas/bitcoin-core-rc-testing/blob/121720-testing-instructions-refinement/docs/v0.21/testing-0.21rc2.md). Tested descriptor wallets, torv3, anchors. No issues.

Tried signet via `bitcoind` and GUI, both failed to connect to any peers. Followed instructions on [Bitcoin Wiki](https://en.bitcoin.it/wiki/Signet). Possibly an issue with my system but I was able to get peers/headers/blocks on mainnet, mainnet+tor, and testnet. Not sure how to troubleshoot further.

**Comment by hebasto** [comment]
In the GUI the detail view shows disconnected peer's info.
See https://github.com/bitcoin-core/gui/issues/158

**Comment by mjdietzx** [comment]
One thing I noticed while testing that doesn't look quite right (I'll provide a summary of all the testing I've done once I finish).

I have a watch-only wallet. Everything works properly. But when I run `getwalletinfo` in `Console` for this wallet:
```json
{  
   "walletname": "NoDeal",  
   "walletversion": 169900,  
   "format": "bdb",  
   "balance": 0.00000000,  
   "unconfirmed_balance": 0.00000000,  
   "immature_balance": 0.00000000,  
   "txcount": 17,  
   // ... //
   "paytxfee": 0.00000000,  
   "private_keys_enabled": false,  
   "avoid_reuse": false,  
   "scanning": false,  
   "descriptors": false
}
```

Everything is correct in the UI, and `txcount` is correct in the returned json response. However, `balance` is incorrect (it is not 0). Since I didn't find any open issues related to this, and I found this while testing v0.21.0 I'm sharing here.

**Comment by mjdietzx** [comment]
I've been testing/using on master, pulling pretty regularly, since v0.21.0rc1. MacBook Air 2020

> macOS Catalina

installed from source (gcc), IBD, basic command line tests, basic gui tests, create encrypted wallet in gui, send and receive from encrypted wallet (reported and fixed 8008ef770f3d0b14d03e22371314500373732143), import watch-only / skeleton wallet, send via PSBT (created in gui, signed on device, broadcasted in gui) and receive to watch-only, testing a bunch of random stuff in GUI (ie viewing peers, simple commands in console), checking recent transactions in the mempool, transaction received notifications work, and general use of wallet

> then I updated to macOS Big Sur v11.0.1

installed from source (both gcc and clang), bunch of random stuff in gui (ie testing keyboard shortcuts for things like masking values), pretty much all the same tests as I list above

Then I installed the tagged release v0.21.0rc3, built from source (clang), and went through this [script](https://github.com/adamjonas/bitcoin-core-rc-testing/blob/121720-testing-instructions-refinement/docs/v0.21/testing-0.21rc2.md) except for the Signet tests (thanks @adamjonas), all worked ✅

**Comment by stackman27** [comment]
0.21.0rc3 compiled from source, Ubuntu 20.04.1 LTS, Windows 10 64bit

 went through this [doc](https://github.com/adamjonas/bitcoin-core-rc-testing/blob/121720-testing-instructions-refinement/docs/v0.21/testing-0.21rc2.md)- descriptor wallets, anchors worked very well

**Comment by Emzy** [comment]
0.21.0rc3 downloaded from https://bitcoincore.org/bin/bitcoin-core-0.21.0/test.rc3/bitcoin-0.21.0rc3-osx.dmg macOS 10.15.7 (19H2)

I was able to run the wrongly signed rc3 binary from the .dmg after setting an extended attribute via:
```
sudo xattr -rd com.apple.quarantine /Applications/Bitcoin-Qt.app
```
went through the Testing Guide all worked well.

**Comment by hebasto** [comment]
Observing a regression in v0.21.0rc3 comparing to v0.20.1: the option `-onlynet=ipv4` (both in command line and in `bitcoin.conf`) does not work as expected.

In v0.21.0rc3 (and in current master) providing `-onlynet=ipv4` does not prevent outgoing Tor connections.

```
$ src/bitcoind -help | grep -A 4 onlynet
  -onlynet=<net>
       Make outgoing connections only through network <net> (ipv4, ipv6 or
       onion). Incoming connections are not affected by this option.
       This option can be specified multiple times to allow multiple
       networks.
```

**Comment by jonatack** [comment]
@hebasto this may possibly be what @Saibato has been describing for some time; see https://github.com/bitcoin/bitcoin/commit/39858289583cc11fdcefbc8e8716ff09855049a7 which I ended up dropping from #19961 because the issue surprised people but it should probably either be fixed or documented.

**Comment by jonatack** [comment]
e.g. see this comment https://github.com/bitcoin/bitcoin/pull/19961#discussion_r544700961. I'm opening a PR for the doc updates that were dropped.

**Comment by hebasto** [comment]
@jonatack 
> @hebasto this may possibly be what @Saibato has been describing for some time; see [3985828](https://github.com/bitcoin/bitcoin/commit/39858289583cc11fdcefbc8e8716ff09855049a7) which I ended up dropping from #19961 because the issue surprised people but it should probably either be fixed or documented.

What changes in the code did cause such a change in the client behavior?

**Comment by jonatack** [comment]
@hebasto I haven't dug into it but see #19358 I think

**Comment by pinheadmz** [comment]
OSX 10.14.6 (Mojave) testing v0.21.0rc3 at 9ae536d36, built from source.

Followed the guide at https://github.com/adamjonas/bitcoin-core-rc-testing/blob/121720-testing-instructions-refinement/docs/v0.21/testing-0.21rc2.md

Made a descriptor wallet, connected to TorV3 peers, synced to default Signet!

`make check` and `test_runner` pass all tests.

**Comment by n1bor** [comment]
Create new wallet in Windows defaults to un-encrypted wallet. Seems odd default.

**Comment by Sjors** [comment]
@n1bor see discussion in https://github.com/bitcoin-core/gui/pull/96

**Comment by Rspigler** [comment]
Compiled (GCC) v0.21.0rc5 in Qubes 4.0 in a Debian 10 StandaloneVM on Purism hardware

Tested descriptor wallet creation, TorV3, Signet (received and sent - fee estimation didn't work?), and Anchor Connections.

**Comment by sipa** [comment]
> Signet (received and sent - fee estimation didn't work?)

Fee estimation can't work without a substantial amount of transactions making it into the mempool and getting confirmed.

**Comment by ellemouton** [comment]
Followed [these instructions](https://github.com/adamjonas/bitcoin-core-rc-testing/blob/121720-testing-instructions-refinement/docs/v0.21/testing-0.21rc2.md). Compiled 0.21.0rc3 from source on OSX 10.14 Mojave.
Tested descriptor wallets, Tor3 and anchors. All went smoothly 😎

**Comment by hebasto** [comment]
Noted a regression in `QTabWidget` on macOS Big Sur 11.1:

- master (3a6acd1772100d71de45355381d0d6ccf1320748):
![Screenshot from 2021-01-07 19-20-22](https://user-images.githubusercontent.com/32963518/103924418-a4c25500-511e-11eb-903f-de5030b7ff78.png)

- gitian build downloaded from https://bitcoincore.org/bin/bitcoin-core-0.21.0/test.rc5/ (a window is focused):
![Screenshot from 2021-01-07 19-21-19](https://user-images.githubusercontent.com/32963518/103924587-ea7f1d80-511e-11eb-94ba-39a52c2fa90e.png)

NOTE: I'm using a VM for testing, so asking other contributors to check this issue on native macOS.

**Comment by jarolrod** [comment]
@hebasto confirming the `QTabWidget` regression on native macOS `Big Sur 11.1`. 
On `Big Sur 11.1`, this bug can be found on the previous release candidates and  even on the 0.20.1 release. I cannot replicate on my macOS `Catalina 10.15.7` machine. Another interesting thing about this bug is that if you click on another window, the text will reappear.

**Confirming Regression (Big Sur 11.1)**
<img width="587" alt="Screen Shot 2021-01-07 at 12 37 15 PM" src="https://user-images.githubusercontent.com/23396902/103926061-6ad55c80-50e6-11eb-9b28-6f25f72a64bf.png">

**Clicking on another Window, Text Reappears (Big Sur 11.1)**
<img width="604" alt="Screen Shot 2021-01-07 at 12 39 03 PM" src="https://user-images.githubusercontent.com/23396902/103926091-758ff180-50e6-11eb-8bc5-073b728a1183.png">

**Comment by lscarrol** [comment]
0.21.0rc5, Manjaro 20.2.1 

Doc is sound, Tested descriptor wallets, torv3, and anchors

**Comment by hebasto** [comment]
Regarding visual issues on Big Sur: https://github.com/bitcoin-core/gui/pull/177.

**Comment by AnthonyRonning** [comment]
macOS Catalina (10.15.5), built tag v0.21.0rc5 from source, upgraded from a previous version I use occasionally on this computer (v0.19.99.0-225aa5d6d)

Followed 0.21 RC Testing Guide, ran through everything except signet - so far so good. 

Only thing I noticed was on first boot was a mempool file problem: 

```
2021-01-08T02:28:56Z loadblk thread start
2021-01-08T02:28:56Z torcontrol thread start
2021-01-08T02:28:56Z Bound to 127.0.0.1:8333
2021-01-08T02:28:56Z Bound to 127.0.0.1:8334
2021-01-08T02:28:56Z init message: Loading P2P addresses...
2021-01-08T02:28:56Z Loaded 48217 addresses from peers.dat  170ms
2021-01-08T02:28:56Z ERROR: DeserializeFileDB: Failed to open file /Volumes/Samsung750/Bitcoin/Mainnet/mainnet/anchors.dat
2021-01-08T02:28:56Z 0 block-relay-only anchors will be tried for connections.
2021-01-08T02:28:56Z init message: Starting network threads...
2021-01-08T02:28:56Z net thread start
2021-01-08T02:28:56Z dnsseed thread start
2021-01-08T02:28:56Z addcon thread start
2021-01-08T02:28:56Z opencon thread start
2021-01-08T02:28:56Z msghand thread start
2021-01-08T02:28:56Z init message: Done loading
2021-01-08T02:28:56Z Waiting 300 seconds before querying DNS seeds.
2021-01-08T02:29:27Z Failed to deserialize mempool data on disk: CAutoFile::read: end of file: unspecified iostream_category error. Continuing anyway.
2021-01-08T02:29:27Z loadblk thread exit
```

But continued just fine and didn't show back up again, maybe a coincidence.

**Comment by pkrasam** [comment]
Hardware: MacBook Pro
OS: macOS Big Sur v11.1 (20C69)
RC: 0.21.0rc5 binary
Tests: Completed
- created a descriptor wallet, 
- connected over Torv3, 
- ran through the anchors test, 
- all of ^ + looking forward to trialling Signet

**Comment by calkob** [comment]
v0.21.0rc3 
Windows 10 
Bitcoin-qt

Had the warning below pop up a few times at the start of testing whilst trying to create wallets.  couldnt find a reproducible scenario to pass on.  all i can think is that this wasnt a fresh build there was already wallets in the wallet file with a db.

all other tests ran fine

![wallet test](https://user-images.githubusercontent.com/16057802/104041595-35f20400-51d1-11eb-8305-2aef0931a3b4.png)

**Comment by Sjors** [comment]
@calkob did you experiment with descriptor wallets before you tested this release?

**Comment by calkob** [comment]
> @calkob did you experiment with descriptor wallets before you tested this release?

I dont think i had run this build before today but now that you mention it i am doubting myself.  Could that be an answer for the warning?

**Comment by Sjors** [comment]
If you've ever used the master branch or any of @achow101's recent wallet pull requests, your `~/.bitcoin/wallets` dir may contain descriptor wallets. Those have been changed a few times, so you might have an old incompatible version lying around.

**Comment by nomo-nomad** [comment]
MacBook Pro running macOS 11.1 Big Sur, testing 0.21rc3 compiled from source

✅ Successfully created a descriptor wallet from qt GUI and in the terminal/CLI
✅ Anchors.dat appears on clean shut down and disappears on node startup
✅ Successfully connected to Torv3

Not sure if I missed a build option -- but bitcoin-qt is showing that it's using Berkeley DB not SQLite

**Comment by scgbckbone** [comment]
@nomo-nomad do you have sqlite installed?

**Comment by scgbckbone** [comment]
acer aspire E15 Intel Core i5-6200U
4.15.0-130-generic #134-Ubuntu SMP Tue Jan 5 20:46:26 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
Ubuntu 18.04
build from source

* Successfully created a descriptor wallet from qt GUI and in the terminal/CLI              **OK**
* Anchors.dat appears on clean shut down and disappears on node startup                 **OK**
* Successfully connected to Torv3                                                                                   **FAIL**


debug.log (during startup):
```
2021-01-11T21:43:24Z torcontrol thread start
2021-01-11T21:43:24Z tor: Authentication cookie /var/run/tor/control.authcookie could not be opened (check permissions)
```
tor version:
```
tor                 0.3.2.10-1     amd64          anonymizing overlay network for TCP
```
no V3 addresses in Peers window (maybe because of ongoing attack?)

i have this in my torrc:
```
HashedControlPassword <n:password>
ControlPort 9051
```
tor socket open (`netstat -nltp`):
```
LISTEN      0            128                     127.0.0.1:9050                   0.0.0.0:*          
LISTEN      0            128                     127.0.0.1:9051                   0.0.0.0:* 
```

**Comment by haegwan** [comment]
Tested based on @jarolrod's [guide](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.21-Release-Candidate-Testing-Guide). It was very easy to follow. Thanks!

Hardware: MacBook Air (M1, 2020)
OS: macOS Big Sur v11.1 (20C69)
Scope: Only a descriptor wallet creation. Other tests follow. 

As others noted, noticed interace glitch both on Light/Dark mode for GUI.
https://github.com/bitcoin-core/gui/pull/177

<img width="781" alt="Screen Shot 2021-01-12 at 6 59 52" src="https://user-images.githubusercontent.com/668076/104244714-5c03f680-54a6-11eb-80c9-ab805bba852e.png">
<img width="804" alt="Screen Shot 2021-01-12 at 6 53 53" src="https://user-images.githubusercontent.com/668076/104244722-60c8aa80-54a6-11eb-90e6-a587cb541980.png">

I also tested Japanese - most texts showed fine. Here are few updates I'd make.
- "current block height" shown as English
- Date format shown as English.

Not sure why it's showing up in this way as all JP translations should be already resolved on transifex.

<img width="768" alt="Screen Shot 2021-01-12 at 7 19 22" src="https://user-images.githubusercontent.com/668076/104244941-c1f07e00-54a6-11eb-8ac4-a89c5a9269a3.png">

**Comment by jarolrod** [comment]
@haegwan this is a known issue with `bitcoin-qt` on macOS Big Sur 11.1 built with `Qt 5.9.8`. It would be useful if you could patch [GUI #177](https://github.com/bitcoin-core/gui/pull/177) (as you linked to it) on top of 0.21rc5 built with `Qt 5.9.8` and report back if it fixed the visual glitches for you. 

Note: If you build with a more recent version of Qt you won't see this glitch.

**Comment by robot-dreams** [comment]
Also tested based on @jarolrod's [guide](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/0.21-Release-Candidate-Testing-Guide), after reading @adamjonas's [post on the importance of testing](https://adamjonas.com/bitcoin/releases/testing/testing-bitcoin-core/). Thanks both of you for the awesome articles!

tl;dr **Every command line test worked perfectly**, didn't test bitcoin-qt

## Hardware / OS

MacBook Pro (Retina, 15-inch, Mid 2015), macOS Catalina Version 10.15.7

## Which release candidate

v0.21.0rc5, compiled from source

Flags for `configure`:

```
--without-miniupnpc
--disable-bench
--without-gui
CXXFLAGS="-O0 -g"
CFLAGS="-O0 -g"
```

## What I tested

I tested everything using an existing data directory (mainnet already synced).

### Descriptor wallets (command line)

```
$ bitcoin-cli -named createwallet wallet_name="v0.21.0-rc5-descriptor-wallet" descriptors=true
{
  "name": "v0.21.0-rc5-descriptor-wallet",
  "warning": "Wallet is an experimental descriptor wallet"
}

$ bitcoin-cli -rpcwallet="v0.21.0-rc5-descriptor-wallet" getwalletinfo
{
  "walletname": "v0.21.0-rc5-descriptor-wallet",
  "walletversion": 169900,
  "format": "sqlite",
  "balance": 0.00000000,
  "unconfirmed_balance": 0.00000000,
  "immature_balance": 0.00000000,
  "txcount": 0,
  "keypoolsize": 3000,
  "keypoolsize_hd_internal": 3000,
  "paytxfee": 0.00000000,
  "private_keys_enabled": true,
  "avoid_reuse": false,
  "scanning": false,
  "descriptors": true
}
```

### Tor v3 (command line)

```
$ cat bitcoin.conf | grep onion
listenonion=1
addnode=sxjbhmhob2xasx3vdsy5ke5j5jwecmh3ca4wbs7wf6sg4g2lm3mbszqd.onion:8333
addnode=rp7k2go3s5lyj3fnj6zn62ktarlrsft2ohlsxkyd7v3e3idqyptvread.onion:8333
addnode=d6jwdcoo2l3gbjps6asgg4nhp2gn5oao3wj333o43ssqnjaliehytfad.onion:8333

$ bitcoin-cli -netinfo 4
Bitcoin Core v0.21.0rc5 - 70016/Satoshi:0.21.0/

Peer connections sorted by direction and min ping
<-> relay   net  mping   ping send recv  txn  blk  age id address                                                             version
...
out  full onion    482    482    2    3    0         4  2 rp7k2go3s5lyj3fnj6zn62ktarlrsft2ohlsxkyd7v3e3idqyptvread.onion:8333 70016/Satoshi:21.99.0(Medea)/
out  full onion    508    730    2    1    0         4  1 sxjbhmhob2xasx3vdsy5ke5j5jwecmh3ca4wbs7wf6sg4g2lm3mbszqd.onion:8333 70016/Satoshi:0.21.0/
out  full onion    510    581    0   13    0         4  3 d6jwdcoo2l3gbjps6asgg4nhp2gn5oao3wj333o43ssqnjaliehytfad.onion:8333 70016/Satoshi:21.99.0(h42)/
...
                    ms     ms  sec  sec  min  min  min
```

### Signet

```
$ cat signet/bitcoin.conf 
signet=1

$ bitcoind -datadir=signet
2021-01-12T22:20:39Z Bitcoin Core version v0.21.0rc5 (release build)
2021-01-12T22:20:39Z Signet derived magic (message start): 0a03cf40
...

$ bitcoin-cli -datadir=signet getnewaddress
tb1qlz...

# Use faucet at https://signet.bc-2.jp/ to send coins

$ bitcoin-cli -datadir=signet getunconfirmedbalance 
0.00100000
```

### Anchors

```
$ ls anchors.dat
anchors.dat

$ bitcoind
2021-01-12T23:14:44Z Bitcoin Core version v0.21.0rc5 (release build)
...
2021-01-12T23:14:56Z Loaded 2 addresses from "anchors.dat"
2021-01-12T23:14:56Z 2 block-relay-only anchors will be tried for connections.

$ ls anchors.dat
ls: anchors.dat: No such file or directory

# Shut down bitcoind
2021-01-12T23:16:48Z DumpAnchors: Flush 2 outbound block-relay-only peer addresses to anchors.dat started
2021-01-12T23:16:48Z DumpAnchors: Flush 2 outbound block-relay-only peer addresses to anchors.dat completed (0.01s)
...
2021-01-12T23:16:48Z Shutdown: done

$ ls anchors.dat
anchors.dat

$ bitcoind
2021-01-12T23:17:48Z Bitcoin Core version v0.21.0rc5 (release build)
...
2021-01-12T23:17:58Z Loaded 2 addresses from "anchors.dat"
2021-01-12T23:17:58Z 2 block-relay-only anchors will be tried for connections.
```

### Bonus: specter-desktop

specter-desktop v1.0.0 is still able connect to the node and load an existing wallet

![specter-desktop](https://user-images.githubusercontent.com/4276679/104389975-8202b880-54f1-11eb-8d4e-63289e46559f.png)

**Comment by robot-dreams** [comment]
Also repeated the command line tests on a [RaspiBolt](https://stadicus.github.io/RaspiBolt/) and **once again everything worked great**!

Hardware / OS: Raspberry Pi 4 Model B Rev 1.1, Raspbian GNU/Linux 10 (buster)
Which release candidate: v0.21.0rc5 binary, [ARM (32 bit)](https://bitcoincore.org/bin/bitcoin-core-0.21.0/test.rc5/bitcoin-0.21.0rc5-arm-linux-gnueabihf.tar.gz)
What I tested: descriptor wallets, Tor v3, signet, anchors

**Comment by hMsats** [comment]
@jonatack 

> I'm happy to addnode to anyone who is testing 0.21 with tor v3 so you can see an inbound connection. Ping me on bitcoin-core-dev IRC.

Yes please: l4l6gkjqj4vodcwwooyk5tpnotec6sbk4g3sz4hrfziaiw3pxlv3h5qd.onion

Thanks in advance (ex sumBTC)

**Comment by jonatack** [comment]
@hMsats done! you should see a tor v3 node in your inbound connections.

**Comment by hMsats** [comment]
I do! Thanks. This begs (if you haven't done so already) for an onionv2 and onionv3 in your -netinfo like for ipv4 and ipv6  :-)

**Comment by jonatack** [comment]
The getpeerinfo#network field that -netinfo reads doesn't distinguish between v2 and v3 onions; see the Network enum in src/netaddress.h and GetNetworkName() in netbase.cpp. 

At any rate, by the next release  (v22.0), tor v2 will be essentially EOL.

**Comment by jonatack** [comment]
(p.s. happy to hear that you are using -netinfo :))

**Comment by hMsats** [comment]
I do, as it gives a really nice overview of all my inbound and outbound connections and type of connections!
